### ISSUE TITLE
Install issue on osx in 4.03 switch

### ISSUE THREAD

**Issue by dbuenzli** [mirage/ocaml-github#127]
Compilation seems to hang on: 

```
/Users/dbuenzli/.opam/4.03.0/bin/ocamlfind ocamlc -c -g -annot -bin-annot -principal -strict-sequence -I lib -I unix -I js -package atdgen -package bytes -package cmdliner -package cohttp -package cohttp.js -package cohttp.lwt -package cohttp.lwt-core -package js_of_ocaml -syntax camlp4o -package js_of_ocaml.syntax -package lwt -syntax camlp4o -package lwt.syntax -package lwt.unix -package stringext -package uri -package yojson -w @5@8@10@11@12@14@23@24@26@29 -I lib_test -I js -I lib -I unix -o lib_test/test_javascript.cmo lib_test/test_javascript.ml
```

**Comment by dbuenzli** [comment]
Here's what was reported when I killed the thing:
```
E: Failure("Command ''/Users/dbuenzli/.opam/4.03.0/bin/ocamlbuild' lib/github_core.cma lib/github_core.cmxa lib/github_core.a lib/github_core.cmxs unix/github_unix.cma unix/github_unix.cmxa unix/github_unix.a unix/github_unix.cmxs js/github_js.cma js/github_js.cmxa js/github_js.a js/github_js.cmxs jar/jar.native jar/create_release.native jar/sync_releases.native jar/upload_release.native jar/list_releases.native jar/list_events.native jar/list_issues.native jar/search.native jar/listen_events.native gist/gist.native lib_test/current_user.native lib_test/get_token.native lib_test/repo_info.native lib_test/issues.native lib_test/tags.native lib_test/milestones.native lib_test/releases.native lib_test/deploy_keys.native lib_test/pulls.native lib_test/create_issue.native lib_test/create_milestone.native lib_test/create_pull.native lib_test/create_hook.native lib_test/delete_all_hooks.native lib_test/create_statuses.native lib_test/organizations.native lib_test/parse_events.native lib_test/test_javascript.byte -tag debug' terminated with error code 100")
```

**Comment by avsm** [comment]
Hm, memory problems?

**Comment by dbuenzli** [comment]
> Hm, memory problems?

WDYM ? This is on my usual machine (also it seems to be tests why does the package them ?).

**Comment by rgrinberg** [comment]
Could it be an old version? I remember fixing the test building issue. I just checked master again and no tests are built.

**Comment by dbuenzli** [comment]
No this happens while trying to install `1.0.0`.

**Comment by rgrinberg** [comment]
Actually, looks like my commit didn't make it into `1.0.0`:

https://github.com/mirage/ocaml-github/commit/5eb03f90a819e1cf21a0295a82c3cb19cff35cb5

Would you mind pinning master and reporting the result?

**Comment by dbuenzli** [comment]
That worked.

**Comment by juanchanco** [comment]
For the record, I was getting the same issue on debian and netbsd. The problem seems to be when js_of_ocaml is installed. Running `opam install github -v` I see the build hang at:

`ocamlfind ocamlc -c -g -annot -bin-annot -principal -strict-sequence -I lib -I unix -I js -package atdgen -package bytes -package cmdliner -package cohttp -package cohttp.js -package cohttp.lwt -package cohttp.lwt-core -package js_of_ocaml -syntax camlp4o -package js_of_ocaml.syntax -package lwt -syntax camlp4o -package lwt.syntax -package lwt.unix -package stringext -package uri -package yojson -w @5@8@10@11@12@14@23@24@26@29 -I lib_test -I js -I lib -I unix -o lib_test/test_javascript.cmo lib_test/test_javascript.ml`

At that point, ocamlc-opt proceeds to eat up _all_ the memory available and eventually the build fails. Pinning to --dev-repo also worked for me to get it successfully installed.

**Comment by dbuenzli** [comment]
Would it be possible to make a release to fix this ? It has been twenty days and you are a dependency of `opam-submit`. This affects anyone wanting to publish OPAM packages using this tool.

**Comment by dsheets** [comment]
Is this not fixed in 1.1.0?

**Comment by dbuenzli** [comment]
Ah yes sorry got caught again about unpinning resulting in the installation of the previous version. Btw. it's also a good idea to close issues once they are resolved.
